hicommonwealth/commonwealth
# Snapshot knock integration

## Link to Issue
Closes: #7686

## Description of Changes
- Moves `processSnapshotProposalCreated` function to the Knock worker
  - Also significantly simplified the implementation
- Created a migration which fixes snapshot subscriptions by ensuring a community_id is passed to `CommunityAlerts` in the trigger function that keeps the `Subscriptions` table in sync with all the new subscription tables.

## Test Plan
- Add `SnapshotProposalCreated` to the `ALLOWED_EVENTS` env var in your `commonwealth/.env` file
- Set the `KNOCK_API_KEY` in your `.env` file (DM me to get the key)
- Run `yarn start-knock` in `packages/commonwealth/`
- In another terminal run `yarn start-message-relayer` in `packages/commonwealth`
- In another terminal run `yarn workspace snapshot start`
- In yet another terminal :p start the commonwealth app
- Go to the `dydx` community and then go to the snapshots page and click the subscribe button
- Note your user id: `SELECT user_id FROM "Addresses" WHERE address = '[your address]';`
- Execute the following curl command to simulate a new snapshot proposal
```
curl -X POST --location "http://localhost:8001/snapshot" \
    -H "Content-Type: application/json" \
    -d '{
          "id": "proposal/0x5ed0465ba58b442f1e671789797d5e36b538a27603549639a34f95451b59ad32",
          "event": "proposal/start",
          "title": "test-title",
          "body": "test-body",
          "choices": ["1", "2"],
          "space": "dydxgov.eth",
          "start": 1710467951,
          "expire": 1710467951,
          "token": "a88a91630b95e5e7ae9cd8610aff862d2b70383926b7236c973eedec46e0de65",
          "secret": "a88a91630b95e5e7ae9cd8610aff862d2b70383926b7236c973eedec46e0de65"
        }'
```
- Now, join the common-dev Knock workspace (just create an account with your `@common.xyz` email)
- Go to https://dashboard.knock.app/common-dev/development/developers/logs/runs and look for a workflow with the recipient as the user id noted above and the workflow as 'snapshot-proposal'
- Once you find the workflow run, click on it and then click on the `In-App` step and then click on the hyperlinked 'message' link. Once there, switch to the 'Preview' tab and make sure the snapshot notification appears as it should

## Deployment Plan

1.

## Other Considerations

-
